init/pkg/update_pod_status_util.go

```diff
@@ -0,0 +1,842 @@
+
+					dirtyVolumes = remove(dirtyVolumes, volume.volumeName)
+				}(blkdiscard)
+				log.Println("INFO: Command submitted", "volume", volume)
+
+			default:
+				return dirtyVolumes, fmt.Errorf("invalid effective_wipe_method %s", volume.effectiveWipeMethod)
+			}
+		}
+	}
+
+	close(guard)
+	wg.Wait()
+	log.Println("INFO: All cleanup jobs finished successfully")
+
+	return dirtyVolumes, nil
+}
+
+func wipeVolumes(podName string, aeroCluster *asdbv1beta1.AerospikeCluster, dirtyVolumes []string) ([]string, error) {
+	var wg sync.WaitGroup
+
+	nsDevicePaths, nsFilePaths, err := getNamespaceVolumePaths(podName, aeroCluster)
+	if err != nil {
+		return dirtyVolumes, fmt.Errorf("failed to get namespaced volume paths %v", err)
+	}
+
+	rack, err := getRack(podName, aeroCluster)
+	if err != nil {
+		return dirtyVolumes, fmt.Errorf("failed to get rack of pod %s %v", podName, err)
+	}
+
+	workerThreads := rack.Storage.CleanupThreads
+	guard := make(chan struct{}, workerThreads)
+
+	persistentVolumes := getPersistentVolumes(getAttachedVolumes(rack, aeroCluster))
+	for volIndex := range persistentVolumes {
+		vol := &persistentVolumes[volIndex]
+		if vol.Aerospike == nil {
+			continue
+		}
+
+		volume := newVolume(podName, vol)
+		switch volume.volumeMode {
+		case BlockVolumeMode:
+			if utils.ContainsString(nsDevicePaths, volume.aerospikeVolumePath) {
+				log.Println("INFO: Wiping volume", "volume", volume)
+
+				if _, err := os.Stat(volume.getMountPoint()); err != nil {
+					return dirtyVolumes, fmt.Errorf("mounting point %s does not exist %v", volume.getMountPoint(), err)
+				}
+
+				switch volume.effectiveWipeMethod {
+				case DDMethod:
+					stderr, err := os.CreateTemp("/tmp", "init-stderr")
+					if err != nil {
+						return dirtyVolumes, err
+					}
+
+					dd := []string{DDMethod, "if=/dev/zero", "of=" + volume.getMountPoint(), "bs=1M"}
+
+					wg.Add(1)
+					guard <- struct{}{}
+
+					go func(cmd []string) {
+						wg.Done()
+
+						if err := execute(cmd, stderr); err != nil {
+							dat, err := os.ReadFile(stderr.Name())
+							if err != nil {
+								stderr.Close()
+								os.Remove(stderr.Name())
+								panic(err.Error())
+							}
+
+							if !strings.Contains(string(dat), "No space left on device") {
+								panic(err.Error())
+							}
+						}
+
+						stderr.Close()
+						os.Remove(stderr.Name())
+						log.Println("INFO: Execution completed", "cmd", cmd)
+						<-guard
+
+						dirtyVolumes = remove(dirtyVolumes, volume.volumeName)
+					}(dd)
+					log.Println("INFO: Command submitted", "volume", volume)
+
+				case BLKDiscardMethod:
+					blkdiscard := []string{BLKDiscardMethod, volume.getMountPoint()}
+
+					wg.Add(1)
+					guard <- struct{}{}
+
+					go func(cmd []string) {
+						wg.Done()
+
+						if err := execute(cmd, nil); err != nil {
+							panic(err.Error())
+						}
+
+						log.Println("INFO: Execution completed", "cmd", cmd)
+						<-guard
+
+						dirtyVolumes = remove(dirtyVolumes, volume.volumeName)
```

Comment:
Same can be done here also. Maybe create 2 func like `runDD` and `runBLkDiscard`